includes/class-sensei-quiz.php

```diff
@@ -1280,17 +1280,45 @@ public static function start_quiz_questions_loop() {
 		global $sensei_question_loop;
 
 		// Initialise the questions loop object.
-		$sensei_question_loop['current']   = -1;
-		$sensei_question_loop['total']     = 0;
-		$sensei_question_loop['questions'] = array();
+		$sensei_question_loop['current']        = -1;
+		$sensei_question_loop['total']          = 0;
+		$sensei_question_loop['questions']      = [];
+		$sensei_question_loop['posts_per_page'] = -1;
+		$sensei_question_loop['current_page']   = 1;
+
+		// Get the pagination settings and populate the loop object.
+		$pagination = json_decode(
+			get_post_meta( get_the_ID(), '_pagination', true ),
+			true
+		);
 
-		$questions = Sensei()->lesson->lesson_quiz_questions( get_the_ID(), 'publish' );
+		if ( ! empty( $pagination['pagination_number'] ) ) {
+			$sensei_question_loop['posts_per_page'] = (int) $pagination['pagination_number'];
+		}
 
-		if ( count( $questions ) > 0 ) {
-			$sensei_question_loop['total']     = count( $questions );
-			$sensei_question_loop['questions'] = $questions;
-			$sensei_question_loop['quiz_id']   = get_the_ID();
+		// phpcs:ignore WordPress.Security.NonceVerification -- Argument is used for pagination in the frontend.
+		if ( ! empty( $_GET['questions-page'] ) ) {
+			// phpcs:ignore WordPress.Security.NonceVerification
+			$sensei_question_loop['current_page'] = max( 1, (int) $_GET['questions-page'] );
+		}
+
+		// Fetch the questions.
+		$all_questions = Sensei()->lesson->lesson_quiz_questions( get_the_ID(), 'publish' );
+		if ( ! $all_questions ) {
+			return;
 		}
+
+		// Paginate the questions.
+		if ( $sensei_question_loop['posts_per_page'] > 0 ) {
+			$offset              = $sensei_question_loop['posts_per_page'] * ( $sensei_question_loop['current_page'] - 1 );
+			$paginated_questions = array_slice( $all_questions, $offset, $sensei_question_loop['posts_per_page'] );
+		} else {
+			$paginated_questions = $all_questions;
+		}
+
+		$sensei_question_loop['total']     = count( $all_questions );
```

Comment:
Here we need all the questions count in order to render the pagination in the front-end.